sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedSources/HttpRequestDataFeedSource.cs

```diff
@@ -43,7 +43,7 @@ internal HttpRequestDataFeedSource(HttpRequestParameter parameter)
 
             Parameter = parameter;
 
-            Url = new Uri(parameter.Url);
+            Url = parameter.Url == null ? null : new Uri(parameter.Url);
```

Comment:
Is there a test covering this change?

Reply:
Not yet, but we have an issue to add validation as a Viewer (which covers this change): https://github.com/Azure/azure-sdk-for-net/issues/17767

We still need to find a way to test that, though, since we can't test a newly created data feed as a viewer since we're automatically assigned the admin role. Related: https://github.com/Azure/azure-sdk-for-net/issues/17765